org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js

```diff
@@ -13,101 +13,202 @@ function validateArgumentType(arg, argName, expectedType) {
+    }
+
+    // Build the oic.r.dimming resource
+    var dim = {
+        "href": "/dim",
+        "rt": ["oic.r.light.dimming"],
+        "if": ["oic.if.a", "oic.if.baseline"]
+    }
+
+    // Include the values is expand is specified
+    if (expand) {
+        power.id = 'power';
+        power.value = providerSchema['attributes'].switch == 'on';
+
+        dim.id = 'dim';
+        dim.dimmingSetting = providerSchema['attributes'].level;
+        dim.range = [0, 100];
+    }
 
     return {
-        id: deviceSchema['id'],
-        n: deviceSchema['name'],
-        rt: 'org.opent2t.sample.lamp.superpopular',
-        power: { 'value': deviceSchema['attributes'].switch == 'on' },
-        dim: { 'dimmingSetting': deviceSchema['attributes'].level, 'range': [0, 100] }
+        opent2t: {
+            schema: 'org.opent2t.sample.lamp.superpopular',
+            translator: 'opent2t-translator-com-smartthings-lightbulb',
+            controlId: deviceId
+        },
+        pi: providerSchema['id'],
+        mnmn: providerSchema['manufacturer'],
+        mnmo: providerSchema['model'],
+        n: providerSchema['name'],
+        rt: ['org.opent2t.sample.lamp.superpopular'],
+        entities: [
+            {
+                rt: ['opent2t.d.light'],
+                di: providerSchema['id'],
+                icv: 'core.1.1.0',
+                dmv: 'res.1.1.0',
+                resources: [
+                    power,
+                    dim
+                ]
+            }
+        ]
     };
 }
 
-// Helper method to convert the translator schema to the device schema.
-function translatorSchemaToDeviceSchema(translatorSchema) {
+/***
+ * Converts an OCF platform/resource schema for calls to the SmartThings API
+ */
+function resourceSchemaToProviderSchema(resourceId, resourceSchema) {
 
     // build the object with desired state
     var result = {};  
 
-    if (translatorSchema.power) {
-        result['switch'] = translatorSchema.power.value? 'on' : 'off';
+    switch (resourceId) {
+        case 'power':
+            result['switch'] = resourceSchema.value ? 'on' : 'off';
+            break;
+        case 'dim':
+            result['level'] = resourceSchema.dimmingSetting;
+            break;
+        default:
+            // Error case
+            throw new Error("Invalid resourceId");
     }
-
-    if (translatorSchema.dim) {
-        result['level'] = translatorSchema.dim.dimmingSetting;
-    }
-
     return result;
 }
 
 var deviceId;
 var smartthingsHub;
 
-// This translator class implements the 'org.opent2t.sample.lamp.superpopular' interface.
+// This translator class implements the 'opent2t.d.light' interface.
 class Translator {
 
     constructor(deviceInfo) {
         console.log('SmartThings Lightbulb initializing...');
 
         validateArgumentType(deviceInfo, "deviceInfo", "object");
         
-        deviceId = deviceInfo.deviceInfo.id;
+        deviceId = deviceId = deviceInfo.deviceInfo.opent2t.controlId;
```

Comment:
Fixed.

---
In reply to: [90384829](https://github.com/openT2T/translators/pull/134#discussion_r90384829) [](ancestors = 90384829)